src/Solver.cpp

```diff
@@ -489,52 +655,212 @@ void Solver::initializeCmfd(){
+ *          (e.g., an eigenvalue calculation) in addition to any user-defined
+ *          fixed sources.
+ *
+ *          This method may be called by the user to compute the scalar flux 
+ *          for a fixed source distribution from Python as follows:
+ *
+ * @code
+ *          // Assign fixed sources
+ *          // ...
+ * 
+ *          // Find the flux distribution resulting from the fixed sources
+ *          solver.computeFlux(max_iters=100)
+ * @endcode
+ *
+ *          Alternatively, as described above, this method may be called by
+ *          the user in Python to compute the flux from a superposition of
+ *          fixed and / or eigenvalue sources as follows:
+ *
+ * @code
+ *          // Solve for sources and scalar flux distribution
+ *          solver.computeEigenvalue(max_iters=1000)
+ *
+ *          // Add fixed source(s)
+ *          // ...
+ *          
+ *          // Find fluxes from superposition of eigenvalue and fixed sources
+ *          solver.computeFlux(max_iters=100, only_fixed_source=False)
+ * @endcode
+ *
+ *
+ * @param max_iters the maximum number of source iterations to allow
+ * @param only_fixed_source use only fixed sources (true by default)
  */
-void Solver::checkTrackSpacing() {
+void Solver::computeFlux(int max_iters, bool only_fixed_source) {
+
+  if (_track_generator == NULL)
+    log_printf(ERROR, "The Solver is unable to compute the flux "
+               "since it does not contain a TrackGenerator");
 
-  int* FSR_segment_tallies = new int[_num_FSRs];
-  int num_segments;
-  segment* curr_segment;
-  segment* segments;
-  Cell* cell;
+  log_printf(NORMAL, "Computing the flux...");
 
-  /* Set each tally to zero to begin with */
-  #pragma omp parallel for
-  for (int r=0; r < _num_FSRs; r++)
-    FSR_segment_tallies[r] = 0;
+  /* Clear all timing data from a previous simulation run */
+  clearTimerSplits();
 
-  /* Iterate over all azimuthal angles, all tracks, and all Track segments
-   * and tally each segment in the corresponding FSR */
-  #pragma omp parallel for private (num_segments, curr_segment)
-  for (int i=0; i < _tot_num_tracks; i++) {
+  /* Start the timer to record the total time to converge the flux */
+  _timer->startTimer();
 
-    num_segments = _tracks[i]->getNumSegments();
-    segments = _tracks[i]->getSegments();
+  FP_PRECISION residual;
 
-    for (int s=0; s < num_segments; s++) {
-      curr_segment = &segments[s];
-      FSR_segment_tallies[curr_segment->_region_id]++;
+  /* Initialize data structures */
+  initializePolarQuadrature();
+  initializeExpEvaluator();
+
+  /* Initialize new flux arrays if a) the user requested the use of 
+   * only fixed sources or b) no previous simulation was performed which
+   * initialized and computed the flux (e.g., an eigenvalue calculation) */
+  if (only_fixed_source || _num_iterations == 0) {
+    initializeFluxArrays();
+    flattenFSRFluxes(0.0);
+  }
+
+  initializeSourceArrays();
+  initializeFSRs();
+  zeroTrackFluxes();
```

Comment:
Great catch, I didn't think I needed it here at the time I implemented this method. But then I went on to add the different convergence criteria, and I see that now it is necessary to count fissionable FSRs in case the criterion is ``FISSION_SOURCE``.